rasa_core/policies/ensemble.py

```diff
@@ -113,10 +117,13 @@ def persist(self, path):
         # type: (Text) -> None
         """Persists the policy to storage."""
 
-        self._persist_metadata(path, self.max_history())
+        self._persist_metadata(path, self._max_histories())
 
-        for policy in self.policies:
-            policy.persist(path)
+        for i, policy in enumerate(self.policies):
+            # TODO better way then many folders?
+            # TODO delete old files from these folders
```

Comment:
is there a reason not to use one folder? if there is an issue with that multiple folders are fine.

Reply:
this is the fastest solution, i came with to avoid renaming feauturizers, because we have multiple ones now

Reply:
then we would need to clean the folder before writing, because if you changed policy, old files will be there, and it looks confusing

Reply:
also, if you have two keras policies with different featurizer, I think it is clearer to keep them in different folders

Reply:
ok sounds good

Reply:
I think agent should clear model folder before writing into it, could you please recommend a good way to do it?

Reply:
yes that is a good idea. 

`shutil.rmtree(model_dir)` but there should be a check if that folder really contains a model (e.g. check if `policy_metadata.json` is present or something like that)